portality/static/doaj/js/doaj.facetview.theme.js

```diff
@@ -133,22 +245,32 @@ function doajScrollTop(options, context) {
     });
 }
 
-function doajPostRender(options, context) {
-    doajScrollTop(options, context);
-
+function doajToggleAbstract(options, context) {
     // toggle the abstracts
     $('.abstract_text', context).hide();
     $(".abstract_action", context).unbind("click").click(function(event) {
         event.preventDefault();
         var el = $(this);
-        //var text = el.html();
-        //var newText = text == "(expand)" ? "(collapse)" : "(expand)";
-        //el.html(newText);
         $('.abstract_text[rel="' + el.attr("rel") + '"]').slideToggle(300);
         return true;
     });
 }
 
+function doajPostRender(options, context) {
+    doajScrollTop(options, context);
+    doajToggleAbstract(options, context);
+
+    // Update the widget options & generated text
+    if (options.sharesave_link) {
+        var widget_text = doajGenFixedQueryWidget(options);
+        $('.facetview_embedwidget', context).val(widget_text);
+    }
+}
+
+function doajFixedQueryWidgetPostRender(options, context) {
```

Comment:
:+1: very nice use of the 2 different functions to achieve different results with normal integration and the widget. fv2 seems to have gotten a lot of code structure right to allow custom f() to be written in this particular manner. Don't know what you're missing till you don't have it (with fv1).